### ISSUE TITLE
Leah Needs Help :) -- A Few Questions

### ISSUE THREAD

**Issue by lwasser** [earthlab/hub-ops#49]
Ok @betatim  first and foremost this is so amazing. i'm in the hub and it's working
ok so on to questions and things to work out.

1. Currently this hub is linked to a repo, here: https://github.com/earthlab-education/2018-07-20-spatial-python-workshop/ it drops the stuff from that repo into a materials folder. 
QUESTION: How do I 
* change the name of the folder "materials" to something else?
* let's say i add some notebooks to it today, it doesn't seem to update. At the same time if i add a notebook that a student might edit, what would happen if i update that notebook? what i'm wondering is if we need to make a materials folder that is either not editable that the students then make a copy to start working in? 
2. I have a similar question about the data. Currently it drops the data in home/earth-analytics/data/ this is so perfect! however... same question as above. let's say the student starts working in that folder. but i want to add a new dataset to it. Example - i just added 2 new datasets yesterday and updated earthpy. How does that work? should we always have students save stuff outside of the data and what is now called materials directory? 

This is so so great... going to play more.

**Comment by betatim** [comment]
> change the name of the folder "materials" to something else?

The target directory is [an argument to nbgitpuller](https://github.com/earthlab/hub-ops/blob/c92ed5ed4da4032af349424dc1e7fdc051935a40/wshub/values.yaml#L40)

> i add some notebooks to it today, it doesn't seem to update.

If you add a notebook to `master` and stop and start your server (or logout and back in) then the contents of `materials` should get updated. `nbgitpuller` runs every time your single user notebook server starts up.

`nbgitpuller` will update files that have not been edited by a student. If they have been edited it won't update them (assumption is the student wants to keep those changes). If a student wants to "start from scratch" they can delete that notebook and restart their server.

> updating the data

[these lines](https://github.com/earthlab/hub-ops/blob/c92ed5ed4da4032af349424dc1e7fdc051935a40/wshub/values.yaml#L36-L39) check if there is a folder called `earth-analytics/data/spatial-vector-lidar` and if not copies over the original data. It doesn't look at it on a file by file basis.

> i just added 2 new datasets yesterday and updated earthpy

Right now it doesn't fetch all of the datasets that earthpy knows about. Only the lidar one [via this line](https://github.com/earthlab/hub-ops/blob/c92ed5ed4da4032af349424dc1e7fdc051935a40/user-images/wshub/Dockerfile#L12). We can duplicate this to fetch more datasets, or eventually build a small function as part of earthpy that fetches all datasets.

> how do I know that we are using a new image on the hub (and hence should see my changes)?

Right now that is hard to tell. We could/should create a file called something like `/dockerimagename` that contains the name and the tag of the docker image you are using. Then you could look at that from within the hub. To see which version you are getting. If you like the idea we should make a separate issue so it doesn't get forgotten.

**Comment by betatim** [comment]
> doc docs docs!?

Yes! I have been adding `README`s all around the repo and started `day-to-day.md` but right now it isn't organised well enough yet. The idea of `day-to-day` was to collect short instructions on things like changing versions, adding a new hub, short command snippets for debugging, etc

**Comment by lwasser** [comment]
ok @betatim there are a few things not working. The most important thing is the data. i updated the env as we discussed. Then i updated the hub-ops to the most recent build. But my data do not update. they are still the original data that i had when i logged in this  morning. i can't delete the data dir in jupyter because it says the directory has files in it.Can you see if you can get the data to update to the most recent version of earthpy? it doesn't seem to update the data when i update earthpy as we discussed. we need the newest data for the workshop tomorrow

2. the second big issue is that no one can actually use the hub except me (my admin works) but the other accounts do not work. you'll see a bunch of screenshots in the #jupyterhub channel. 
Here is one: we need this to work for our workshop. 
[<img width="1126" alt="screen shot 2018-07-19 at 4 42 41 pm" src="https://user-images.githubusercontent.com/7649194/42974273-c309124c-8b73-11e8-8dfb-f90272a634bc.png">
](url)

i'm hoping you know what is happening :) i didn't think to test other accounts earlier today after seeing my account work just fine. the data thing also is very weird. i don't know why it won't update the data download once it downloads it EVEN THOUGH i did update earthpy... OH yes i do. `replace=False` that function looks for the folder and will only replace it if this argument is set. BUT i renamed the directory and it still didn't redownload. ... not sure what is up!
